libminifi/include/core/ContentRepository.h

```diff
@@ -30,13 +31,23 @@
 #include "ContentSession.h"
 #include "core/RepositoryMetricsSource.h"
 #include "core/Core.h"
+#include "core/StreamAppendLock.h"
 
 namespace org::apache::nifi::minifi::core {
 
 /**
  * Content repository definition that extends StreamManager.
  */
 class ContentRepository : public core::CoreComponent, public StreamManager<minifi::ResourceClaim>, public utils::EnableSharedFromThis<ContentRepository>, public core::RepositoryMetricsSource {
+  class ContentStreamAppendLock : public StreamAppendLock {
```

Comment:
it reflects the class hierarchy of ContentRepository and StreamManager, since StreamManager generates the write stream (which are possibly append streams) locking for append should appear at that level

Reply:
I meant line 42, ContentStreamAppendLock : StreamAppendLock

Reply:
yes, that is what I replied to